signing/signer.go

```diff
@@ -0,0 +1,58 @@
+package signing
+
+import (
+	"errors"
+	"github.com/spacemeshos/ed25519"
+	"github.com/spacemeshos/go-spacemesh/log"
+)
+
+type BlsFetcher interface {
+	FetchBlsKey(edKey []byte) []byte
+}
+
+type ActivationValidator interface {
+	IsActive(blsKey []byte) bool
+}
+
+type Signer interface {
+	Sign(m []byte) []byte
+	Verifier() Verifier
+}
+
+type EdSigner struct {
+	privKey ed25519.PrivateKey // the pub & private key
+	pubKey  ed25519.PublicKey  // only the pub key part
+}
+
+func NewEdSignerFromBuffer(buff []byte) (*EdSigner, error) {
+	if len(buff) < 32 {
+		return nil, errors.New("buffer too small")
+	}
+	return &EdSigner{privKey: buff, pubKey: buff[:32]}, nil
```

Comment:
Ask @barakshani if there is any way to validate that the private key matches the public key

Reply:
I am not sure what is the context here, but if you hold the private key, you can always recompute the public key from it

Reply:
I sign and verify.

Reply:
@barakshani 's solution is more elegant and efficient